src/InteractiveBridge.ts

```diff
@@ -207,82 +226,183 @@ export class InteractiveBridge extends LiveryBridge {
-        const strValue = stringify(value);
-        throw new Error(
-          `subscribeOrientation value: ${strValue}, should be: "landscape" | "portrait"`,
-        );
-      }
-      return value;
-    }
-
+  subscribeOrientation(listener: (value: Orientation) => void) {
     return this.sendCommand('subscribeOrientation', undefined, (value) =>
-      listener(validate(value)),
-    ).then(validate);
+      listener(orientation.parse(value)),
+    ).then((value) => orientation.parse(value));
+  }
+
+  /**
+   * Returns promise of current playback paused state
+   * and calls back `listener` with any subsequent paused state updates.
+   */
+  async subscribePaused(listener: (value: boolean) => void) {
+    const state = await this.subscribePlaybackState((value) =>
+      listener(pausedState.safeParse(value).success),
+    );
+    return pausedState.safeParse(state).success;
+  }
+
+  /**
+   * Returns promise of current LiveryPlayer playback state
+   * and calls back `listener` with any subsequent state updates.
+   */
+  subscribePlaybackState(listener: (value: PlaybackState) => void) {
+    return this.sendCommand('subscribePlaybackState', undefined, (value) =>
+      listener(playbackState.parse(value)),
+    ).then((value) => playbackState.parse(value));
+  }
+
+  /**
+   * Returns promise of current playback playing state
+   * and calls back `listener` with any subsequent playing state updates.
+   */
+  async subscribePlaying(listener: (value: boolean) => void) {
+    const state = await this.subscribePlaybackState((value) =>
+      listener(playingState.safeParse(value).success),
+    );
+    return playingState.safeParse(state).success;
+  }
+
+  /**
+   * Returns promise of current LiveryPlayer playback qualities
+   * and calls back `listener` with any subsequent qualities changes.
+   */
+  subscribeQualities(listener: (value?: Qualities) => void) {
+    return this.sendCommand('subscribeQualities', undefined, (value) =>
+      listener(qualities.parse(value)),
+    ).then((value) => qualities.parse(value));
   }
 
   /**
    * Returns promise of current LiveryPlayer playback quality
    * and calls back `listener` with any subsequent quality changes.
+   *
+   * @deprecated Instead use {@link subscribeQualities}.active
    */
   subscribeQuality(listener: (value: string) => void) {
-    function validate(value: unknown) {
-      if (typeof value !== 'string') {
-        throw new Error(
-          `subscribeQuality value type: ${typeof value}, should be: string`,
-        );
-      }
-      return value;
-    }
-
     return this.sendCommand('subscribeQuality', undefined, (value) =>
-      listener(validate(value)),
-    ).then(validate);
+      listener(validateString(value)),
+    ).then(validateString);
+  }
+
+  /**
+   * Returns promise of current playback stalled state
+   * and calls back `listener` with any subsequent stalled state updates.
+   */
+  async subscribeStalled(listener: (value: boolean) => void) {
```

Comment:
See subscribePaused above; please use`reducedSubscribe()` and a simple reducer function here.